examples/upgrade-http1x.md

```diff
@@ -0,0 +1,645 @@
+
+## Before You Begin
+
+**Important Considerations:**
+
+1. **Breaking Changes**: This is a major version upgrade with breaking API changes
+2. **Testing Required**: Thoroughly test your application after migration
+3. **Gradual Migration**: Consider migrating one service at a time
+4. **Legacy Examples**: The `examples/legacy/` directory contains fully working http@0.2 examples for reference
+
+**Compatibility:**
+- Minimum Rust version: Check your generated SDK's rust-toolchain.toml
+- Tokio runtime: Still compatible with tokio 1.x
+- Tower middleware: Compatible with tower 0.4
+
+## Dependency Updates
+
+### Cargo.toml Changes
+
+#### Server Dependencies
+
+**Before (http@0.2/hyper@0.14):**
+```toml
+[dependencies]
+http = "0.2"
+hyper = { version = "0.14.26", features = ["server"] }
+tokio = "1.26.0"
+tower = "0.4"
+
+# Server SDK with legacy HTTP support
+pokemon-service-server-sdk = {
+    path = "../pokemon-service-server-sdk/",
+    package = "pokemon-service-server-sdk-http0x",
+    features = ["request-id"]
+}
+
+[dev-dependencies]
+hyper = { version = "0.14.26", features = ["server", "client"] }
+hyper-rustls = { version = "0.24", features = ["http2"] }
+aws-smithy-legacy-http = { path = "../../../rust-runtime/aws-smithy-legacy-http/" }
+```
+
+**After (http@1.x/hyper@1.x):**
+```toml
+[dependencies]
+http = "1"
+hyper = { version = "1", features = ["server"] }
+hyper-util = { version = "0.1", features = ["tokio", "server", "server-auto", "service"] }
+tokio = { version = "1.26.0", features = ["rt-multi-thread", "macros"] }
+tower = "0.4"
+
+# Server SDK with http@1.x support
+pokemon-service-server-sdk = {
+    path = "../pokemon-service-server-sdk/",
+    features = ["request-id"]
+}
+
+[dev-dependencies]
+bytes = "1"
+http-body-util = "0.1"
+hyper = { version = "1", features = ["server", "client"] }
+hyper-util = { version = "0.1", features = ["client", "client-legacy", "http1", "http2"] }
+hyper-rustls = { version = "0.27", features = ["http2"] }
+aws-smithy-runtime = { path = "../../rust-runtime/aws-smithy-runtime" }
+```
+
+**Key Changes:**
+- `http`: `0.2` → `1`
+- `hyper`: `0.14.26` → `1`
+- **New**: `hyper-util` crate for server and client utilities
+- **New**: `bytes` and `http-body-util` for body handling
+- `hyper-rustls`: `0.24` → `0.27`
+- Server SDK package name: no longer needs `-http0x` suffix
+- Replace `aws-smithy-legacy-http` with `aws-smithy-runtime`
+
+#### Client Dependencies
+
+**Before:**
+```toml
+[dependencies]
+aws-smithy-runtime = { version = "1.0", features = ["client"] }
+aws-smithy-runtime-api = "1.0"
+```
+
+**After:**
+```toml
+[dependencies]
+aws-smithy-http-client = "1.0"  # Replaces direct hyper usage
+aws-smithy-runtime = { version = "1.0", features = ["client"] }
+aws-smithy-runtime-api = "1.0"
+```
+
+## Server-Side Changes
+
+### 1. Import Changes
+
+**Before:**
+```rust
+use hyper::StatusCode;
+```
+
+**After:**
+```rust
+use http::StatusCode;
```

Comment:
can we re-export this from the server or `aws-smithy-http-server`? I don't see any compelling reason to force a direct `http` dep.

Reply:
It is being re-exported from the aws-smithy-http-server: rust-runtime/aws-smithy-http-server/src/lib.rs.

Need to change the guide.